File: frontend/src/lib/api/schemas/research.ts
import { z } from "zod";

export const researchDocumentReadSchema = z.object({
  id: z.string().uuid(),
  project_id: z.string().uuid(),
  source_url: z.string(),
  title: z.string(),
  content: z.string(),
  citation: z.record(z.string(), z.unknown()),
  qdrant_point_id: z.string().nullable(),
  published_at: z.string().nullable(),
});
export type ResearchDocumentRead = z.infer<typeof researchDocumentReadSchema>;

export type ResearchDocumentCreateRequest = {
  source_url: string;
  title: string;
  content: string;
  citation?: Record<string, unknown>;
  published_at?: string | null;
};

File: backend/src/ytforge/interfaces/api/schemas/research.py
from __future__ import annotations

import uuid
from datetime import datetime
from typing import Any

from pydantic import BaseModel, ConfigDict, Field


class ResearchDocumentCreateRequest(BaseModel):
    source_url: str = Field(min_length=1, max_length=2048)
    title: str = Field(min_length=1, max_length=500)
    content: str
    citation: dict[str, Any] = Field(default_factory=dict)
    published_at: datetime | None = None


class ResearchDocumentRead(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: uuid.UUID
    project_id: uuid.UUID
    source_url: str
    title: str
    content: str
    citation: dict[str, Any]
    qdrant_point_id: str | None
    published_at: datetime | None
